src/Orleans/AssemblyLoader/AssemblyProcessor.cs

```diff
@@ -75,7 +75,20 @@ public void Initialize()
                 Assembly[] assemblies = AppDomain.CurrentDomain.GetAssemblies();
 
                 // initialize serialization for already loaded assemblies.
-                CodeGeneratorManager.GenerateAndCacheCodeForAllAssemblies();
+                var generated = CodeGeneratorManager.GenerateAndCacheCodeForAllAssemblies();
```

Comment:
I believe this is the only place that uses `GenerateAndCacheCodeForAllAssemblies`. If that is indeed the case, just kill that method and pass the collection of assemblies that you just got from the shim, so we reduce the shim's usage.